pkg/apis/migration/v1alpha1/directimagemigration_types.go

```diff
@@ -89,6 +92,20 @@ func (r *DirectImageMigration) GetDestinationCluster(client k8sclient.Client) (*
 	return GetCluster(client, r.Spec.DestMigClusterRef)
 }
 
+// Get the MigMigration that owns this DirectImageMigration. If not owned, return nil.
+func (r *DirectImageMigration) GetOwner(client k8sclient.Client) (*MigMigration, error) {
+	owner := &MigMigration{}
+	ownerRefs := r.GetOwnerReferences()
+	if len(ownerRefs) > 0 {
+		ownerRef := types.NamespacedName{Name: ownerRefs[0].Name, Namespace: r.Namespace}
```

Comment:
why the first owner ref? are we not looking for a specific resource?

Reply:
I think we're doing this because we only ever create a single owner reference, so if there are more than one, then someone has manually hacked the resource into an unsupported scenario anyway. But yes, we should probably check the type of the owning resource to make sure it's what we expect. But even then, what do we do if we find more than one? Do we take the first, or to we error out? (this is a general question which might apply elsewhere. There are various cases where it is *possible* to find 2 resources in a slice but we only really expect one, so we generally take the first and ignore the rest. Perhaps we should always error out if we find more than one when we expect one?)

Reply:
+1 I'll check the type of the ownerRef. In the current system there is never more than one owner but since that could change, it would be good to support the future case.

Reply:
Fixed.